### PR TITLE
Source name validation

### PR DESCRIPTION
**Pull request by admanesachin**
### Fixed Issues (if relevant)

1. magento-engcom/msi#<issue_number>: Issue title
2. ...

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/inventory?pullRequest=2359) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/inventory?pullRequest=2359) before we can accept your contribution.<br/>

File: app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php [hunk 1]
```diff
@@ -41,6 +41,8 @@ public function validate(SourceInterface $source): ValidationResult
             $errors[] = __('"%field" can not be empty.', ['field' => SourceInterface::SOURCE_CODE]);
         } elseif (preg_match('/\s/', $value)) {
             $errors[] = __('"%field" can not contain whitespaces.', ['field' => SourceInterface::SOURCE_CODE]);
+        } elseif (preg_match('/\$[:]*{(.)*}/', $value)) {
+            $errors[] = __('"%field" can not contain invalid characters.', ['field' => SourceInterface::SOURCE_CODE]);
```
Comment (nathanjosiah) [inline app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php]: Does a translation for this exist already?

Comment (admanesachin) [reply app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php]: I don't think. Is there a more generic one that i should use instead? wasn't sure what the recommended way is.

Comment (admanesachin) [reply app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php]: I can do that. Should i add it in the i18n/en_us.csv file for that module?

Comment (nathanjosiah) [reply app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php]: For a patch release we can't add new translations. If the next release isn't a patch then this is fine.

Comment (admanesachin) [reply app/code/Magento/Inventory/Model/Source/Validator/CodeValidator.php]: Updated the message in the latest commit.

**Pull request merged**